docs/scripting-ref.rst

```diff
@@ -2230,3 +2230,126 @@ Trigger mouse release event in web page.
 **Async:**: no.
 
 See notes about mouse events in :ref:`splash-mouse-click`.
+
+.. _splash-send-keys:
+
+splash:send_keys
+----------------
+
+Send keyboard events to page context.
+
+**Signature:** ``splash:send_keys(keys)``
+
+**Parameters**
+
+* keys - string representing the keys to be sent as keyboard events.
+
+**Returns:** nil
+
+**Async:**: no.
+
+Key sequences are specified by using a small subset of emacs edmacro syntax:
+
+* whitespace is ignored and only used to separate the different keys
+* characters are literally represented
+* words within brackets represent function keys, like ``<Return>``, ``<Left>``
+  or ``<Home>``. See `Qt docs`__ for a full list of function keys. ``<Foo>``
+  will try to match ``Qt::Key_Foo``.
+
+__ http://doc.qt.io/qt-5/qt.html#Key-enum
+
+Following table shows some examples of macros and what they would generate on
+an input:
+
+======================    ===============
+Macro                     Result
+======================    ===============
+``Hello World``           ``HelloWorld``
+``Hello <Space> World``   ``Hello World``
+``< S p a c e >``         ``<Space>``
+``Hello <Home> DEL``      ``ello``
+``Hello <Backspace>``     ``Hell``
+======================    ===============
+
+Key events are not propagated immediately until event loop regains control,
+thus :ref:`splash-wait` must be called to reflect the events.
+
+.. _Qt key-enum: http://doc.qt.io/qt-5/qt.html#Key-enum
+
+.. _splash-send-text:
+
+splash:send_text
+----------------
+
+Send text as keyboard events to page context.
+
+**Signature:** ``splash:send_text(text)``
+
+**Parameters:**
+
+* text - string to be sent as keyboard events.
+
+**Returns:** nil
+
+**Async:**: no.
+
+Key events are not propagated immediately until event loop regains control,
+thus :ref:`splash-wait` must be called to reflect the events.
+
+This function is to be used to directly send text input as keyboard events.
+Whilst the macro syntax on :ref:`splash-send-keys` is useful for sending
+function keys, on some cases direct text input is more appropriate. It
+completely bypasses macro definition and sends text as character keys.
+
+This function in conjuction with :ref:`splash-send-keys` and covers most needs
+on keyboard input, such as filling in forms and submitting them.
+
+Examples:
+
+Example 1: focus first input, fill in a form and submit
+
+.. code-block:: lua
+
+    function main(splash)
+        assert(splash:go(splash.args.url))
+        assert(splash:wait(0.5))
+        splash:send_keys("<Tab>")
+        splash:send_text("zerocool")
+        splash:send_keys("<Tab>")
+        splash:send_text("hunter2")
```

Comment:
Let's maybe add spaces to `send_text` examples here and to commented-out `send_keys` example below in case someone misses the table in `send_keys` doc.